### PR TITLE
Tord hanoi oop

### PR DESCRIPTION
**Pull request by Kazads**

### CODE REVIEW

File: OOHanoiTord/src/js/Hanoi.js [hunk 1]
```diff
@@ -0,0 +1,67 @@
...
+ topDiskSize(pegIndex) {
+   if (this.pegs[pegIndex].length === 0){
+     return Number.MAX_VALUE;
+   }
+   var peg = this.pegs[pegIndex]
+   return peg[peg.length-1];
+ }
+
+ moveDisk(from, to) {
+   // 1-FROM ROD, 2-USING ROD, 3-TO ROD
```
Comment (KimSvenSand) [inline OOHanoiTord/src/js/Hanoi.js:63]: Remove this comment

File: OOHanoiTord/src/js/Hanoi.js [hunk 2]
```diff
@@ -0,0 +1,67 @@
...
+  towerOfHanoi(numberOfDisks){
+  // direction of rotation of the smallest disk
+  var dir = 0;
+  if(numberOfDisks % 2 == 0){
+    dir = 1;
+  }else{
+    dir = -1;
+  }
+  //CHANGE moves and Pegs to global variables;
+  var pegs = [[], [], []];
```
Comment (KimSvenSand) [inline OOHanoiTord/src/js/Hanoi.js:17]: Remove this

File: OOHanoiTord/src/js/Hanoi.js [hunk 3]
```diff
@@ -0,0 +1,67 @@
...
+        if (this.topDiskSize(next[pegMin]) > this.topDiskSize(prev[pegMin])) {
+          this.moveDisk(prev[pegMin], next[pegMin]);
+        }
+        else {
+          this.moveDisk(next[pegMin], prev[pegMin]);
+        }
+    }
+    moveSmallest = !moveSmallest;
+  }
+  //console.log("Number of moves: "+numberOfMoves);
```
Comment (KimSvenSand) [inline OOHanoiTord/src/js/Hanoi.js:49]: Remove this comment

File: OOHanoiTord/src/js/Hanoi.js [hunk 4]
```diff
@@ -0,0 +1,67 @@
...
+  var next = [1, 2, 0];
+  var prev = [2, 0, 1];
+
+  var moveSmallest = true;
+
+  for (i = 0; i < numberOfMoves; i++) {
+      if (moveSmallest) {
+        var oldPegMin = pegMin;
+        // in JS -1 % 3 === -1, we add 3 to get positive result
+        pegMin = (oldPegMin + dir + 3) % 3;
```
Comment (KimSvenSand) [inline OOHanoiTord/src/js/Hanoi.js:36]: pegMin might be a misleading name

**Review by KimSvenSand — CHANGES_REQUESTED** [review]
Perhaps structure a bit more with classes and objects. Comment some more to make it more clear what the algorithm does. :octocat:

**Review by emilfolino — COMMENTED** [review]
Ser bra ut.

**Pull request closed without merging**